src/linux/agent/system-server.js

```diff
@@ -100,17 +101,20 @@ rpc.exports = {
 
       var intent = packageManager.getLaunchIntentForPackage(pkg);
       if (intent === null && activity === null)
-        throw new Error("Unable to find a front-door activity");
+        throw new Error('Unable to find a front-door activity');
 
       if (intent === null) {
         intent = Intent.$new();
-        intent.setAction("android.intent.action.MAIN");
+        intent.setAction('android.intent.action.MAIN');
       }
 
       if (activity !== null) {
+        var activities = [];
         var pkgInfo = packageManager.getPackageInfo(pkg, GET_ACTIVITIES);
-        var activities = pkgInfo.activities.value.map((activityInfo) => activityInfo.name.value);
-        if (activities.indexOf(activity) < 0)
+        pkgInfo.activities.value.forEach(function (activityInfo) {
+          activities.push(activityInfo.name.value);
+        });
```

Comment:
Sorry, I wasn't clear – `map()` is available, only the arrow function part needed changing.

Reply:
of course